### PR TITLE
Fix hlint issues

### PR DESCRIPTION
**Pull request by markus1189**
Hey, I ran hlint and fixed all the issues.  

There was one about duplication in `App.hs` which I *ignored*, maybe you want to have a look and check if you agree: 

it flags the following three lines:

https://github.com/cdepillabout/termonad/blob/e69e7ac5e8ce96cd5d4db020c6732a534f5781a6/src/Termonad/App.hs#L333-L335

But extracting that to a function which returns all arguments (they are needed further down) seemed like unnecessary to me

### CODE REVIEW

File: src/Termonad/Types.hs [hunk 1]
```diff
@@ -464,7 +462,7 @@ defaultTMConfig =
 -- termonad's behaviour in order to implement new functionality. Fields should
 -- have sane @Semigroup@ and @Monoid@ instances so that config extensions can
 -- be combined uniformly and new hooks can be added without incident.
-data ConfigHooks = ConfigHooks {
+newtype ConfigHooks = ConfigHooks {
```
Comment (cdepillabout) [inline src/Termonad/Types.hs:465]:
```suggestion
data ConfigHooks = ConfigHooks {
```

File: src/Termonad/Term.hs [hunk 2]
```diff
@@ -184,18 +184,16 @@ termExitWithConfirmation tab mvarTMState = do
...
+  join $ modifyMVar mvarTMState $ \tmState -> do
+    let notebook = tmStateNotebook tmState
+        detachTabAction =
+          notebookDetachTab
+            (tmNotebook notebook)
+            (tmNotebookTabTermContainer tab)
+    let newTabs = deleteFL tab (tmNotebookTabs notebook)
+    let newTMState =
+          set (lensTMStateNotebook . lensTMNotebookTabs) newTabs tmState
+    pure (newTMState, detachTabAction)
```
Comment (cdepillabout) [inline src/Termonad/Term.hs:196]: Please revert this change.

**Comment by cdepillabout** [comment]
I'm not a huge fan of `hlint`.  However, most of these changes seem good.

Could you take `hlint` out of CI?  Or, ideally, make it so that `hlint` runs, but it doesn't cause the build to fail?

File: src/Termonad/Term.hs (hunk 2 above)
Comment (markus1189) [reply src/Termonad/Term.hs]: Done!

**Comment by markus1189** [comment]
Okay I see, I removed it from the workflow again

**Comment by cdepillabout** [comment]
@markus1189 I fixed the problem with stack building the examples in dec72e7. 

Are you happy with this PR?  Can I go ahead and merge it in?

**Comment by markus1189** [comment]
Yep thanks!

**Comment by cdepillabout** [comment]
Thanks!
